src/common/device_helpers.cuh

```diff
@@ -402,57 +403,69 @@ struct XGBDefaultDeviceAllocatorImpl : XGBBaseDeviceAllocator<T> {
   }
   void deallocate(pointer ptr, size_t n) {  // NOLINT
     GlobalMemoryLogger().RegisterDeallocation(ptr.get(), n * sizeof(T));
-    return SuperT::deallocate(ptr, n);
+    SuperT::deallocate(ptr, n);
   }
 #if defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
   XGBDefaultDeviceAllocatorImpl()
     : SuperT(rmm::mr::get_current_device_resource(), cudaStream_t{nullptr}) {}
 #endif  // defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
 };
 
-#if defined(XGBOOST_USE_RMM) && XGBOOST_USE_RMM == 1
-template <typename T>
-using XGBCachingDeviceAllocatorImpl = XGBDefaultDeviceAllocatorImpl<T>;
-#else
 /**
- * \brief Caching memory allocator, uses cub::CachingDeviceAllocator as a back-end and logs
- * allocations if verbose. Does not initialise memory on construction.
```

Comment:
This note is still useful,  I bumped into bugs because `dh::caching_device_vector` doesn't initialize memory.

Reply:
Fixed.